packages/now-client/src/types.ts

```diff
@@ -5,6 +5,20 @@ export interface Dictionary<T> {
   [key: string]: T;
 }
 
+export type DeploymentEventType =
+  | 'hashes-calculated'
+  | 'file-count'
+  | 'file-uploaded'
+  | 'all-files-uploaded'
+  | 'created'
+  | 'building'
+  | 'ready'
+  | 'alias-assigned'
+  | 'warning'
+  | 'error'
+  | 'notice'
+  | 'tip';
```

Comment:
The problem with this solution is that you have defined each string twice.

I think it would be best to create an array as the source of truth so they are defined once like so:

```ts
const EVENTS_ARRAY = [
  // File events
  'hashes-calculated',
  'file-count',
  'file-uploaded',
  'all-files-uploaded',
  // Deployment events
  'created',
  'building',
  'ready',
  'alias-assigned',
  'warning',
  'error',
  'notice',
  'tip',
] as const;
const EVENTS = new Set(EVENTS_ARRAY);
type EventsTuple = typeof EVENTS_ARRAY;
type DeploymentEventType = EventsTuple[number];
```

Reply:
TIL 👍